Moving astro vs jsx comparison (formerly chart) and resulting fallout (#1188)
withastro/docs: src/pages/en/core-concepts/astro-components.md

**Astro components** are the basic building blocks of any Astro project. They are HTML-only templating components with no client-side runtime.

**If you know HTML, you already know enough to write your first Astro component.**

> Astro component syntax is a superset of HTML. The syntax was [designed to feel familiar to anyone with experience writing HTML or JSX](#astro-vs-jsx), and adds support for including components and JavaScript expressions. You can spot an Astro component by its file extension: `.astro`.

Author: ```suggestion
Astro component syntax is a superset of HTML. The syntax was [designed to feel familiar to anyone with experience writing HTML or JSX](#differences-between-astro-and-jsx), and adds support for including components and JavaScript expressions. You can spot an Astro component by its file extension: `.astro`.
```

<p class:list={["add", "dynamic", {classNames: true}]} />
```

## JSX-like Expressions

You can can define local JavaScript variables inside of the frontmatter component script within an Astro component. You can then inject these variables into the component's HTML template using JSX-like expressions!

> ### `.astro` vs `.jsx`
> 
> Astro component syntax is a superset of HTML. It was designed to feel familiar to anyone with HTML or JSX experience.
> 
> **If you know HTML, you already know enough to write your first Astro component.**
> 
> | Feature                      | Astro | JSX  |
> | ---------------------------- | ----- | --------- |
> | File extension               | `.astro` | `.jsx` or `.tsx` |
> | User-Defined Components      | `<Capitalized>` | `<Capitalized>`  |
> | Expression Syntax            | `{}` | `{}` |
> | Spread Attributes            | `{...props}` | `{...props}` |
> | Boolean Attributes           | `autocomplete` === `autocomplete={true}` | `autocomplete` === `autocomplete={true}` |
> | Inline Functions             | `{items.map(item => <li>{item}</li>)}`  | `{items.map(item => <li>{item}</li>)}` |
> | Conditional Rendering             | `{condition &&  <p>text<p>}`  | `{condition &&  <p>text<p>}` |
> | IDE Support                  | [VS Code (incl. Open VSX), Nova](/en/editor-setup/) | Phenomenal |
> | Requires JS import           | No    | Yes, `jsxPragma` (`React` or `h`) must be in scope |
> | Fragments                    | Automatic top-level, `<Fragment>` or `<>` inside functions | Wrap with `<Fragment>` or `<>` |
> | Multiple frameworks per-file | Yes | No |
> | Modifying `<head>`           | Just use `<head>` in top-level pages | Per-framework (`<Head>`, `<svelte:head>`, etc) |
> | Comment Style                | `<!-- HTML -->` | `{/* JavaScript */}`  |
> | Special Characters           | `&nbsp;`  | `&nbsp;`  |
> | Attributes                   | `dash-case` | `camelCase`|
> 
> 
> ### Variables

Reviewer: Remove the table in favour of the more detailed but less overwhelming version suggested. Do we still want this “If you know HTML…” line somewhere though? Leaving it here for now but maybe even move right up to somewhere more prominent?

```suggestion
**If you know HTML, you already know enough to write your first Astro component.**

### Variables
```

<Fragment set:html={htmlString} />
```

### Differences between Astro and JSX

Astro component syntax is a superset of HTML. It was designed to feel familiar to anyone with HTML or JSX experience, but there a couple of key differences between `.astro` files and JSX.

> | Feature                      | Astro | JSX  |
> | ---------------------------- | ----- | --------- |
> | Multiple frameworks per-file | Yes | No |
> | Requires JS import           | No    | Yes, `jsxPragma` (`React` or `h`) must be in scope |
> | Attributes                   | `dash-case` | `camelCase`|
> | Modifying `<head>`           | Just use `<head>` in top-level pages | Per-framework (`<Head>`, `<svelte:head>`, etc) |
> | Comment Style                | `<!-- HTML -->` | `{/* JavaScript */}`  |

Author: ```suggestion
```

> 
> | Feature                      | Astro | JSX  |
> | ---------------------------- | ----- | --------- |
> | Multiple frameworks per-file | Yes | No |
> | Requires JS import           | No    | Yes, `jsxPragma` (`React` or `h`) must be in scope |
> | Attributes                   | `dash-case` | `camelCase`|
> | Modifying `<head>`           | Just use `<head>` in top-level pages | Per-framework (`<Head>`, `<svelte:head>`, etc) |
> | Comment Style                | `<!-- HTML -->` | `{/* JavaScript */}`  |

Author: ```suggestion
```

<Fragment set:html={htmlString} />
```

> ## Component Props

Author: Nice, @delucis! I committed this to work from, and will pop the streamlined chart we had been discussing on Discord underneath it, so we have everything to work from. Maybe a smaller chart of only key differences will be a nice visual summary after this explanation? Maybe we'll nix it entirely? I'll put it together and see what it looks like.
